### PR TITLE
FilterVariantTranches documentation fix and improvement

### PR DESCRIPTION
**Pull request by lucidtronix**

### CODE REVIEW

**Comment by droazen** [comment]
@lucidtronix Could you nominate a reviewer? Is this something you want in for this week's release?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java [hunk 1]
```diff
@@ -26,7 +26,13 @@
 import picard.cmdline.programgroups.VariantFilteringProgramGroup;
 
 /**
- * Apply tranche filtering to VCF based on scores from an annotation in the INFO field.
+ * Apply tranche filtering to VCF based on scores from an annotation in the INFO field.  Tranches are specified in
+ * percent sensitivity to the variants in the resource files.  For example, if you specify INDEL tranches 98.0 and 99.0
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: I appreciate the clarity re:variants *in the resource files*!

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java [hunk 2]
```diff
@@ -26,7 +26,13 @@
...
+ * using the CNN_2D score the filtered VCF will contain 2 filter tranches for INDELS: CNN_2D_INDEL_Tranche_98.00_99.00
+ * and CNN_2D_INDEL_Tranche_99.00_100.00. We expect CNN_2D_INDEL_Tranche_99.00_100.00 to be more sensitive but less precise
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: You don't have a 100+ like VQSR?  (where 100+ is for stuff that scores worse than the worst of the resource variants)

Also, you should be explicit that anything scoring better than the indel tranche 98 is PASS.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java [hunk 3]
```diff
@@ -51,10 +60,38 @@
...
+ * This will result in a VCF with filters: CNN_2D_SNP_Tranche_99.00_99.99, CNN_2D_SNP_Tranche_99.99_100
+ * CNN_2D_INDEL_Tranche_98.00_99.00, CNN_2D_INDEL_Tranche_99.00_100
+ * <pre>
+ * gatk FilterVariantTranches \
+ *   -V input.vcf.gz \
+ *   --resource hapmap.vcf \
+ *   --resource mills.vcf \
+ *   --info-key CNN_2D \
+ *   --snp-tranche 99.0 --snp-tranche 99.99 \
+ *   --indel-tranche 98.0 --indel-tranche 99.0 \
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: Do you want to update these recommendations based on Brian's results?  Or note that this is an example and not necessarily "best practices"?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java (hunk 2 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: Added note that all other variants pass.  We put the worse than the worst variants into the _100.00 tranche too, which I guess should be called the 100.00+.  Do you think we should we add the plus sign?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java (hunk 3 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: Added a note that the tranches may need to be tweaked. I want to do an F1 score analysis over a larger region before updating the recomendations. The first test I did  to get these numbers was just over 10m BP in chromosome 20.

**Comment by lucidtronix** [comment]
@ldgauthier Thanks for the review! back to you.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java [hunk 4]
```diff
@@ -29,9 +29,13 @@
  * Apply tranche filtering to VCF based on scores from an annotation in the INFO field.  Tranches are specified in
  * percent sensitivity to the variants in the resource files.  For example, if you specify INDEL tranches 98.0 and 99.0
  * using the CNN_2D score the filtered VCF will contain 2 filter tranches for INDELS: CNN_2D_INDEL_Tranche_98.00_99.00
- * and CNN_2D_INDEL_Tranche_99.00_100.00. We expect CNN_2D_INDEL_Tranche_99.00_100.00 to be more sensitive but less precise
- * than CNN_2D_INDEL_Tranche_98.00_99.00, because variants in CNN_2D_INDEL_Tranche_99.00_100.00 have lower scores than
- * variants in the tranche CNN_2D_INDEL_Tranche_98.00_99.00.
+ * and CNN_2D_INDEL_Tranche_99.00_100.00. Variants that scored better than the 98th percentile of variants in the
+ * resources pass through the filter and will have `.` in the filter field. We expect variants in the tranche
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: I didn't realize your passing variants are dots.  That's not in accordance with the spec:

"PASS if this position has passed all filters, i.e. a call is made at this position"
" If filters have not been applied, then this field should be set to the missing value."
http://samtools.github.io/hts-specs/VCFv4.2.pdf

Comment (cmnbroad) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: Marking these as PASS sounds right, as the output currently makes it look like no filters were applied. The simple change would be to call `passFilters()` in an else block in `secondPassApply`. But that implies that all filters pass, which is fine when we're removing old filters, but thats not the default behavior for the tool. So I think the tool either has to always remove old filters, or the (local filter) PASS case needs to preserve the existing failed filters.

**Comment by lucidtronix** [comment]
@ldgauthier @cmnbroad @bhanugandham Sorry for the delay & mission creep in this PR.  It now addresses #5964 #5939 and annotates with `PASS` thater than `.`.  I also now set the default tranches `99.95` for SNPs and `99.4` for INDELs based on F1 score maximizing experiments from the clinical pipeline, for example:
Screenshot 2019-06-04 08 09 41
 
Rebased and ready for review.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariants.java [hunk 5]
```diff
@@ -59,6 +59,10 @@
  * {@link CNNVariantWriteTensors} and {@link CNNVariantTrain}.
  * CNNVariantTrain will create a json architecture file and an hd5 weights file, which you can use with this tool.
  *
+ * The advanced argument `info-annotation-keys` is available for models trained with different sets info field annotations.
+ * In order to do this you must first train your own model with the tools {@link CNNVariantWriteTensors} and {@link CNNVariantTrain}.
+ * Otherwise, providing this argument with anything but the standard set of annotations will result in an error.
```
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariants.java]: Is there any way to make the Python error more informative?  `ValueError: Error when checking input: expected annotations to have shape (7,) but got array with shape (6,)` from #5939 isn't so clear.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java [hunk 6]
```diff
@@ -51,10 +70,40 @@
  *   --resource hapmap.vcf \
  *   --resource mills.vcf \
  *   --info-key CNN_1D \
- *   --tranche 99.9 --tranche 99.0 --tranche 95 \
+ *   --snp-tranche 99.9 \
```
[the remark this answers is not in the document]
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: Good point fixed. (Except the example of specifying multiple tranches where the point is that they don't need to use the defaults.)

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java [hunk 7]
```diff
@@ -27,6 +27,22 @@
 
 /**
  * Apply tranche filtering to VCF based on scores from an annotation in the INFO field.
+ * The annotation can come from the {@link CNNScoreVariants} tool, VQSR,
```
[the remark this answers is not in the document]
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java]: Added

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariants.java (hunk 5 above)
Comment (lucidtronix) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariants.java]: I added an additional warning on the java side, it's a bit tricky to change the error on the python side without a big refactor.

**Comment by lucidtronix** [comment]
@ldgauthier back to you

**Pull request merged**